skyportal/handlers/api/candidate.py

```diff
@@ -898,13 +903,20 @@ def delete(self, candidate_id):
               application/json:
                 schema: Success
         """
-        c = Candidate.query.get(candidate_id)
-        if not (
-            self.associated_user_object.is_system_admin
-            or c.uploader_id == self.associated_user_object.id
-        ):
-            return self.error("Insufficient permissions.")
-        DBSession().delete(c)
+        cands = (
+            Candidate.query.filter(Candidate.obj_id == obj_id)
+            .filter(Candidate.filter_id == filter_id)
+            .all()
+        )
+        for c in cands:
+            if not (
+                self.associated_user_object.is_system_admin
+                or c.uploader_id == self.associated_user_object.id
+            ):
+                return self.error("Insufficient permissions.")
```

Comment:
Might be useful to state that the error is insufficient permissions for this particular `c`.

Reply:
Done.